### DISCUSSION TITLE
V2.22 | TS101 & S60 Added | PinecilV2 improved

### DISCUSSION THREAD

**Discussion by Ralim** [Ralim/IronOS#1833] [General]
# Headline features

## Sequre S60

The [Sequre S60](https://sequremall.com/products/sequre-s60-nano-electric-soldering-iron-support-pd-qc-power-supply-compatible-with-c210-soldering-iron-tips-precision-electronic-mobile-phone-repair-tool-anti-static-soldering-pen?variant=42361945096380) uses JBC tips, which makes it quite useful for the smaller tip types and extra options available.

## TS101 

The TS101 is the evolution of the TS100, picking up USB-PD.
It has otherwise similar tip support to the TS100/Pinecil/PinecilV2.

Absolutely massive kudos goes to @VioletEternity for her work on the reverse engineering of this. If you at all are helped by IronOS running on this device more credit goes to her than to I. Also big thanks to @whitequark for organising + supporting + magic.

## PinecilV2

1. BLE is fixed on all devices
2. Bootup Logo support is finalised and working
3. Improved the tip control, improving accuracy and remove most oscillations

## Profile heating mode for MHP30

This lets you define a heat profile and run this profile akin to a proper reflow device.
This can be used on the MHP30 by long-holding the A button (aka start button).
Profile can be edited in settings.

## Note on newer OLED's

To prevent this release being held up forever, the TS101 and S60 are being released with a limitation on the OLED screen.
The current code will only draw to the upper left corner of the screen.
Assets have been made for rendering this at full size, but the code is not complete yet.

## Smaller updates

* Filtering added to MHP tilt-exit to make it less sensitive
* Warning if a tip is detected to be shorted (TS101 + PinecilV2)
* Translation updates ❤️
* Documentation updates
* Lots of tooling and code cleanups

## Full change list
* Update translation_CS.json by @cewbdex in https://github.com/Ralim/IronOS/pull/1662
* Add validation ID to BLE by @ew1abz in https://github.com/Ralim/IronOS/pull/1666
* Update translation_FR.json by @HanaO00 in https://github.com/Ralim/IronOS/pull/1667
* MHP30 Soldering/Reflow Temperature Profile by @codingcatgirl in https://github.com/Ralim/IronOS/pull/1672
* Update translation_DE.json by @discip in https://github.com/Ralim/IronOS/pull/1675
* corrected character count by @discip in https://github.com/Ralim/IronOS/pull/1677
* removing extra entries by @discip in https://github.com/Ralim/IronOS/pull/1676
* corrected wording by @discip in https://github.com/Ralim/IronOS/pull/1679
* Update translation_FR.json by @HanaO00 in https://github.com/Ralim/IronOS/pull/1681
* Adjust how heap and ram sections are allocated by @Ralim in https://github.com/Ralim/IronOS/pull/1682
* Update translation_IT.json by @federicodilo in https://github.com/Ralim/IronOS/pull/1685
* Translation_IT.json rev.2 by @federicodilo in https://github.com/Ralim/IronOS/pull/1687
* Merging newest translations into dev by @MAuCTEP in https://github.com/Ralim/IronOS/pull/1689
* Patch 4 by @onxblog in https://github.com/Ralim/IronOS/pull/1691
* S60 Support by @Ralim in https://github.com/Ralim/IronOS/pull/1692
* Unify sha-bangs for python scripts by @ia in https://github.com/Ralim/IronOS/pull/1703
* Update comments by @ia in https://github.com/Ralim/IronOS/pull/1702
* source/Makefile: add HOST_PYTHON, extend clean target, remove -.d generation by @ia in https://github.com/Ralim/IronOS/pull/1701
* AutoStartMode refactoring by @ia in https://github.com/Ralim/IronOS/pull/1712
* source/Makefile: update for -C support by @ia in https://github.com/Ralim/IronOS/pull/1710
* Save settings on change only / All builds OK (#1699) by @ia in https://github.com/Ralim/IronOS/pull/1704
* Show power source type on detailed screen in soldering mode (#1708) by @ia in https://github.com/Ralim/IronOS/pull/1709
* Ts101 by @Ralim in https://github.com/Ralim/IronOS/pull/1695
* Larger OLED Support by @Ralim in https://github.com/Ralim/IronOS/pull/1713
* Spanish translation is 100% complete by @eyebrowdogs in https://github.com/Ralim/IronOS/pull/1715
* Update translation_PT.json by @TheScientistPT in https://github.com/Ralim/IronOS/pull/1716
* Zipping compiler warning about POW_PD_EXT / Two options by @ia in https://github.com/Ralim/IronOS/pull/1711
* A bit of clean-up & update configuration.h files by @ia in https://github.com/Ralim/IronOS/pull/1717
* Adjust divider by @Ralim in https://github.com/Ralim/IronOS/pull/1723
* correct wording for 'max power' by @discip in https://github.com/Ralim/IronOS/pull/1727
* Remove 0.5 ohm increase to resistance by @dallinson in https://github.com/Ralim/IronOS/pull/1728
* Root directory content refactoring by @ia in https://github.com/Ralim/IronOS/pull/1725
* .gitignore: refactor, unify & update by @ia in https://github.com/Ralim/IronOS/pull/1731
* Cosmetic code style refactoring: add enum for screen orientation & define for OLED state by @ia in https://github.com/Ralim/IronOS/pull/1732
* Try side-port of different linker and ram for PinecilV2 by @Ralim in https://github.com/Ralim/IronOS/pull/1730
* Refactoring check for docker to fix a bug to use Makefile inside docker by @ia in https://github.com/Ralim/IronOS/pull/1735
* Update README for new models by @Ralim in https://github.com/Ralim/IronOS/pull/1705
* Fix echo in top-level Makefile for some environments by @ia in https://github.com/Ralim/IronOS/pull/1737
* Suggestion to update BUILD_VERSION policy by @ia in https://github.com/Ralim/IronOS/pull/1733
* Implement autogeneration of Documentation/README.md by @ia in https://github.com/Ralim/IronOS/pull/1738
* Translations/: add missing exec bit for python scripts by @ia in https://github.com/Ralim/IronOS/pull/1742
* clang-format implementation by @ia in https://github.com/Ralim/IronOS/pull/1740
* Documentation/index.md: add Feedback section (Ralim#1552) as suggested by @ia in https://github.com/Ralim/IronOS/pull/1744
* Implement printSymbolDeg() helper function as method for OLED class by @ia in https://github.com/Ralim/IronOS/pull/1743
* Update translation_IT.json by @federicodilo in https://github.com/Ralim/IronOS/pull/1747
* Move check-style core logic from makefile to shell script by @ia in https://github.com/Ralim/IronOS/pull/1745
* Replace /build/source by /build/ironos to eliminate ambiguity with /build/source/source by @ia in https://github.com/Ralim/IronOS/pull/1748
* Move related pieces of documentation from wiki to Documentation/ by @ia in https://github.com/Ralim/IronOS/pull/1746
* Fix github CI clang-format related check_c-cpp job by @ia in https://github.com/Ralim/IronOS/pull/1752
* Implement target in Makefile to run github CI-like checks locally by @ia in https://github.com/Ralim/IronOS/pull/1753
* BLE crash fixing (Part ?) by @Ralim in https://github.com/Ralim/IronOS/pull/1756
* a bit more 'transparent' V calibration by @discip in https://github.com/Ralim/IronOS/pull/1758
* Add Core/Drivers to the scope of style check by clang-format by @ia in https://github.com/Ralim/IronOS/pull/1754
* Simplify commands for jobs in push.yml by @ia in https://github.com/Ralim/IronOS/pull/1761
* push.yml: probable fix #1764 by @ia in https://github.com/Ralim/IronOS/pull/1765
* Decomission of scripts/ci by @ia in https://github.com/Ralim/IronOS/pull/1760
* Update translation_UK.json by @eslng in https://github.com/Ralim/IronOS/pull/1767
* BLE getting started document by @Ralim in https://github.com/Ralim/IronOS/pull/1763
* Remove exec bit from a translation json file by @ia in https://github.com/Ralim/IronOS/pull/1770
* Pinecil v2 short detection at boot by @alextrical in https://github.com/Ralim/IronOS/pull/1729
* Implement ci target for Makefile to reproduce github CI actions & artifacts by @ia in https://github.com/Ralim/IronOS/pull/1769
* Use 3 count filter for MHP30 acceleromter by @Ralim in https://github.com/Ralim/IronOS/pull/1762
* Implement proper git SHA ID marking for builds during pull-requests by @ia in https://github.com/Ralim/IronOS/pull/1772
* Swedish translation update to include new strings by @TheMoo96 in https://github.com/Ralim/IronOS/pull/1779
* metadata.py: update tagging considering github CI builds by @ia in https://github.com/Ralim/IronOS/pull/1782
* Update tagging for detached commit case & update docs by @ia in https://github.com/Ralim/IronOS/pull/1783
* Update translation_IT.json by @federicodilo in https://github.com/Ralim/IronOS/pull/1784
* Update logo flashing information by @Ralim in https://github.com/Ralim/IronOS/pull/1787
* Update build.sh to pass shellcheck checks by @ia in https://github.com/Ralim/IronOS/pull/1789
* Make `make docker-clean` to remove docker build cache by @ia in https://github.com/Ralim/IronOS/pull/1790
* Update translation_CS.json by @davidrothb in https://github.com/Ralim/IronOS/pull/1791
* Update & translate for Bulgarian language by @ia in https://github.com/Ralim/IronOS/pull/1792
* Update & translate for Russian language by @ia in https://github.com/Ralim/IronOS/pull/1793
* Revert "Remove 0.5 ohm increase to resistance" by @Ralim in https://github.com/Ralim/IronOS/pull/1794
* Update Slovak translation by @gamelaster in https://github.com/Ralim/IronOS/pull/1795
* Include Bluetooth document to read-the-docs site generating by @ia in https://github.com/Ralim/IronOS/pull/1798
* Add detailed OLED replacement info to `Troubleshooting` document by @ia in https://github.com/Ralim/IronOS/pull/1797
* Add python packages required by `mkdocs` into docker container by @ia in https://github.com/Ralim/IronOS/pull/1796
* Include History.md to online docs as Changelog by @ia in https://github.com/Ralim/IronOS/pull/1799
* docs/History.md: add info for 2.20 & 2.21 by @ia in https://github.com/Ralim/IronOS/pull/1804
* chore(deps): bump actions/checkout from 3 to 4 by @dependabot in https://github.com/Ralim/IronOS/pull/1811
* Update translation_DE.json by @discip in https://github.com/Ralim/IronOS/pull/1786
* Temperature code updates by @Ralim in https://github.com/Ralim/IronOS/pull/1814
* Add missing Chinese translation by @dogtopus in https://github.com/Ralim/IronOS/pull/1818
* Update Slovak translation by @OndroNR in https://github.com/Ralim/IronOS/pull/1822
* Pinecil v2 tune via PID by @Ralim in https://github.com/Ralim/IronOS/pull/1827

## New Contributors
* @codingcatgirl made their first contribution in https://github.com/Ralim/IronOS/pull/1672
* @MAuCTEP made their first contribution in https://github.com/Ralim/IronOS/pull/1689
* @eyebrowdogs made their first contribution in https://github.com/Ralim/IronOS/pull/1715
* @TheScientistPT made their first contribution in https://github.com/Ralim/IronOS/pull/1716
* @dallinson made their first contribution in https://github.com/Ralim/IronOS/pull/1728
* @alextrical made their first contribution in https://github.com/Ralim/IronOS/pull/1729
* @TheMoo96 made their first contribution in https://github.com/Ralim/IronOS/pull/1779
* @davidrothb made their first contribution in https://github.com/Ralim/IronOS/pull/1791

**Full Changelog**: https://github.com/Ralim/IronOS/compare/v2.21...v2.22

<hr /><em>This discussion was created from the release <a href='https://github.com/Ralim/IronOS/releases/tag/v2.22'>V2.22 | TS101 & S60 Added | PinecilV2 improved</a>.</em>

**Comment by munecito** [comment]
@Ralim 

Any chance that I can invite dinner when I get to Sydney in a couple of weeks to say thank you for getting IronOS to the TS101?

I would take @whitequark and @VioletEternity but I am afraid they are not in Oz.

**Comment by Phixion35** [comment]
Thank you to all for updating and improving IronOS.

Is there any reason I would need to update my Pinecil v2 from v2.21 to this new 2.22 version update?  Thank you again to all!!

**Reply by gamelaster** [reply]
Yes, as on V2.21, there was issue with Bluetooth, so Pinecil did not worked on first try. I recommend to upgrade it. Also, voltage stability is very improved.

**Comment by jockethebastard** [comment]
With PD 3.0, I’m not able to get more than 45W with 20V. Is that how it’s supposed to be, cause I thought it would support 65W? 
Anyways, thanks for doing this, the firmware is super, and very nice to have voltage and wattage on the display.

**Reply by Ralim** [reply]
What device are you referring to? How are you measuring the wattage?
On display wattage is quite approximated + will miss the peak power pulled during shorter re-heat cycles.

**Comment by jockethebastard** [comment]
Sorry I totally forgot to mention what device I was using. It’s a brand new TS-101. I’m only looking at the display, so I guess it’s normal then. /Joakim Norgren8 jan. 2024 kl. 10:46 skrev Ben V. Brown ***@***.***>:﻿
What device are you referring to? How are you measuring the wattage?
On display wattage is quite approximated + will miss the peak power pulled during shorter re-heat cycles.

**Comment by Redwish89** [comment]
Any place to see the progress on a new release? Going to get my TS101 next week but if the fullscreen OLED rendering is coming up i'm think i will hold off a bit on installing IronOS

**Reply by discip** [reply]
You could test the device and as far as I know it does work properly, but as you mentioned the OLED is not yet working as intended.
I recommend waiting for the official release which will support big OLEDs.

**Reply by Redwish89** [reply]
I think i will wait, but if there is a place to keep track on releases it would be nice :)

**Reply by discip** [reply]
You could somewhat track that by watching the evolution of these PRs for example:

https://github.com/Ralim/IronOS/pull/1749
https://github.com/Ralim/IronOS/pull/1819

Or see here whenever a new CI run for the dev branch was performed.
https://github.com/Ralim/IronOS/actions

**Reply by thomasoteixeira** [reply]
@Redwish89 I've been using on my 101. It's worth it even if it doesn't use up all the space available on the screen. I prefer over the stock firmware.
Suggest at least give it a try and rollback to stock if you don't like how it is for now. ;)

**Comment by Jonny-T-cyber** [comment]
I follow the discussion

**Comment by BillyBonito** [comment]
Будет ли прошивка и русский язык на fnirsi HS-01 ?

**Comment by ruslan61rus** [comment]
Is firmware V2.22 for Secure S60, suitable with S60P(new revision)?

**Reply by discip** [reply]
No, but the **`S60P`** version is already being worked on.

**Reply by ruslan61rus** [reply]
Ok, thank  you!

**Comment by DarthZurath** [comment]
TS80P after flashing this firmware shows idle temp at around 170C, which is bizzare because tip is room temp. Needless to say I don't live in hell. Iron also tries QC instead of PD on PD only power supplies and thus stays limited to 5V...
I have no issues with stock firmware, this unfortunately doesn't work as expected for me.

**Reply by Ralim** [reply]
Your likely have one of the newer units with the cloned lower cost STM32 alternative that causes issues with the ADC readings. While these issues should be fixed in 2.23, you are of course welcome to use the vendor firmware :)

**Reply by DarthZurath** [reply]
Thank you for your reply. You're right, 2.23 fixed those issues. Temp readings are still a bit jumpy, but look realistic.

There's also a UI bug I noticed. No matter what I set for boot logo setting has no effect. Every time iron is turned on logo flashes for about half a second and goes to idle screen. Is it a known bug? I couldn't find anything about that in issues. Present in both builds.

**Reply by Ralim** [reply]
Is this a logo you flashed or the one miniware burn into their bootloader? The runtime software can only control our boot logo. It can't adjust the one they put in their bootloader :(

**Reply by DarthZurath** [reply]
It's Miniware logo for sure. That's what I'm talking about: https://github.com/Ralim/IronOS/issues/1308
Do you know if it's possible to find a stock Miniware logo to flash it, so it would look just as stock firmware when it boots.
I can't seem to find stock logo available for download anywhere sadly.

**Reply by Ralim** [reply]
I have no idea on the stock logo sadly. All the collected ones are [saved here](https://github.com/Ralim/IronOS-Meta/tree/main/Bootup%20Logos).

**Reply by DarthZurath** [reply]
Yeah, I've seen those and sadly there's no stock logo there that I can see and the idea is to use stock logo, so transition from bootloader logo to OS logo would look organic.
Do you know maybe if it's possible to extract boot logo from a stock Miniware firmware?
